### PR TITLE
[REF] *: convert AceEditor to owl

### PR DESCRIPTION
**Pull request by aab-odoo**
This PR refactors the website AceEditor to owl. The wrapper
around the lib was defined in web_editor, and extended in website,
where it was used (single usecase). This PR thus introduces an
owl Component (ResourceEditor) to replace it, directly in website,
and specialized to the website usecase.

This thus allows to remove the legacy implementation, and a part
of the legacy compatibility layer (useWidget).

This also removes the last usecase of Widget and select2 library
in the backend bundle, which allows us to trim it down.

Part of task~3439226

### CODE REVIEW

File: addons/website/static/src/components/ace_editor/ace_editor.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+<templates>
+
+    <t t-name="website.CheckableDropdownItem" t-inherit="web.DropdownItem" t-inherit-mode="primary">
+        <xpath expr="//t[@role='menuitem']" position="attributes">
+            <attribute name="role">menuitemcheckbox</attribute>
+            <attribute name="t-att-aria-checked">props.checked ? 'true' : 'false'</attribute>
+        </xpath>
+    </t>
+    
+    <t t-name="website.AceEditor">
```
Comment (kebeclibre) [inline addons/website/static/src/components/ace_editor/ace_editor.xml:11]: (nitpicking)
AceEditor doesn't really use Ace anymore (CodeEditor does). proposals:
AssetsEditor or WebsiteResourcesEditor ?

File: addons/website/static/src/components/ace_editor/ace_editor.js [hunk 2]
```diff
@@ -1,141 +1,501 @@
...
+        this.dialog = useService("dialog");
 
-    //--------------------------------------------------------------------------
-    // Private
-    //--------------------------------------------------------------------------
+        this.keepLast = new KeepLast();
+
+        this.editorRef = useRef("editor");
+
+        this.debug = odoo.debug;
```
Comment (FrancoisGe) [inline addons/website/static/src/components/ace_editor/ace_editor.js:64]: env.debug

**Review by rdeodoo — APPROVED** [review]
As discussed, I just did a functional check, everything seems to be working as it should (multi ressource edit, multi website (COW), errors, prevention overlay etc).
I couldn't properly test the debug mode part (showing key in the select, showing extra scss files etc) as we have an issue ATM that we still need [to fix](https://www.odoo.com/web#id=3475574&menu_id=4720&cids=1&action=333&active_id=974&model=project.task&view_type=form) to correctly forward the debug param.

So it's LGTM functional wise, here is the few minors things I found (but which can be ignored AFAIC or changed later):
- After hitting "reset", the file is correctly reset but the "fa-save" icon is still shown, probably a wrong state being kept?
Note that re-opening the HTML Editor fix this
image

- The indentation and "categories" are a bit harder to notice / see now IMHO, but that can totally be improved later I guess.
See before:
image
After:
image

- The templates are now listed by their name and not their key anymore, which was nice to figure exactly the template you were editing / looking for.
It was also the only way right now to be able to see the view key in the "ace" editor as we have an issue with the debug mode which was the other way to see the key (in debug mode it should appear next to the name).
image
image

**Comment by aab-odoo** [comment]
Thanks a lot for the testing @rdeodoo 

> * After hitting "reset", the file is correctly reset but the "fa-save" icon is still shown, probably a wrong state being kept?

Fixed

> * The indentation and "categories" are a bit harder to notice / see now IMHO, but that can totally be improved later I guess.

I didn't find a trivial way to indent sources inside groups. But the whole SelectMenu could benifit from a design pass. I think that when we introduced it, we basically tried to mock the design of select2. We could indeed ask the designers to look at it in the future.

> * The templates are now listed by their name and not their key anymore, which was nice to figure exactly the template you were editing / looking for.

Fixed

File: addons/web/static/src/legacy/utils.js [hunk 3]
```diff
@@ -41,26 +31,6 @@ export function createWidgetParent(env) {
     };
 }
 
-export function useWidget(refName, widgetClass, params = []) {
-    const ref = useRef(refName);
-    const env = useEnv();
```
Comment (kebeclibre) [inline addons/web/static/src/legacy/utils.js:46]: change in another commit, and the test removal too.

**Pull request closed without merging**